### ISSUE TITLE
Regarding pre_handel

### ISSUE THREAD

**Issue by OliviaZhi** [RunlongYu/BPR_MPR#3]
Hi,

1) I'm wondering, shoud
$$self.predict_ = pre_handel(user_ratings_train, self.predict_, self.item_count)$$
be after the auc_score?

2) You define the scores function right? But I don't see it in your code.

Thanks in advance.

**Comment by EdisonLeeeee** [comment]
> 
> 
> Hi,
> 
>     1. I'm wondering, shoud
>        $$self.predict_ = pre_handel(user_ratings_train, self.predict_, self.item_count)$$
>        be after the auc_score?
> 
>     2. You define the scores function right? But I don't see it in your code.
> 
> 
> Thanks in advance.

 @OliviaZhi , Hi, maybe I could answer your question:

Q1: `self.predict_ = pre_handel(user_ratings_train, self.predict_, self.item_count)`
shouldn't be after the `auc_score`, because the method `pre_handel` convert the scores of users rated in training set to zero, so they will not affect the `auc_score` (cause they always higher than others)

Q2: I didn't see scores function, too. Maybe @RunlongYu doesn't upload it. I can guess it must implemented like this: calculate the top-k (k=5 in his code) rated items for each user, and compare to the test set, which could be used to calculate `precision@k`, `recall@k`, `NDCG@k` and so on.
